### PR TITLE
MANTA-4919 update rebalancer to use generic object from sharkspotter

### PR DESCRIPTION
**Pull request by rjloura**

### CODE REVIEW

File: src/moray_client.rs [hunk 1]
```diff
@@ -53,24 +54,46 @@ pub fn get_manta_object_shark(
...
-    object: &MantaObject,
+    object: &Value,
     etag: &str,
 ) -> Result<(), Error> {
     let mut opts = ObjectMethodOptions::default();
-    let key = object.key.as_str();
-    let value = serde_json::to_value(object)?;
+    let key = match object.get("key") {
+        Some(k) => match serde_json::to_string(k) {
+            Ok(ky) => ky.replace("\"", ""),
```
Comment (rhb2) [inline src/moray_client.rs]: This just strips off the quotes on either side of the string I take it?

Comment (rjloura) [reply src/moray_client.rs]: Yeah.  The string was coming in pre-quoted.  Scratched my head for a couple hours on that one!

File: src/jobs/evacuate.rs [hunk 2]
```diff
@@ -1124,16 +1156,21 @@ impl UpdateMetadata for EvacuateJob {
     /// updates for successful objects.
     fn update_object_shark(
```
[the remark this answers is not in the document]
Comment (rjloura) [reply src/jobs/evacuate.rs]: > I'm approving this. I cited a few places where we may want to eventually move a more common location, minimally for other types of rebalance actions if not for other shark spotter consumers, but I think you're as aware of that as I am and we agree on why it's currently done the way it is.

I completely agree.  I tried to at least factor out what I think will be common functions going forward.  I think our first order of business once this is wrapped up is to look at how we want to refactor things.  I have been thinking about it a lot and have some general ideas. Fortunately I don't think there is TOO much that is tightly coupled.

File: src/jobs/evacuate.rs [hunk 3]
```diff
@@ -1283,12 +1345,79 @@ impl ProcessAssignment for EvacuateJob {
...
+        error: Some(EvacuateObjectError::BadMorayObject),
+        ..Default::default()
+    };
+
+    job_action
+        .insert_into_db(&eobj)
+        .expect("Error inserting bad EvacuateObject into DB");
+}
+
+fn _get_sharks_from_value(
```
[the remark this answers is not in the document]
Comment (rjloura) [reply src/jobs/evacuate.rs]: refactored

File: src/jobs/evacuate.rs [hunk 4]
```diff
@@ -1283,12 +1345,79 @@ impl ProcessAssignment for EvacuateJob {
...
+                "Missing sharks array",
+            )
+            .into());
+        }
+    };
+    serde_json::from_value(sharks_array.to_owned()).map_err(Error::from)
+}
+
+#[allow(non_snake_case)]
+fn _get_objectId_from_value(manta_object: &Value) -> Result<ObjectId, Error> {
```
[the remark this answers is not in the document]
Comment (rjloura) [reply src/jobs/evacuate.rs]: refactored

File: src/jobs/evacuate.rs [hunk 5]
```diff
@@ -73,6 +74,27 @@ struct UpdateEvacuateObject<'a> {
     id: &'a str,
 }
 
+#[derive(Deserialize)]
```
[the remark this answers is not in the document]
Comment (rjloura) [reply src/jobs/evacuate.rs]: Agreed.  I added a comment.

**Pull request merged**